CMakeLists.txt

```diff
@@ -22,6 +22,8 @@ INCLUDE(GetGitRevisionDescription)
 INCLUDE(CheckLibraryExists)
 INCLUDE(CheckLibcxxAtomic)
```

Comment:
I'm experimenting with cross-compiling core using gitian. It seems that the `CheckLibcxxAtomic` include must be moved further down, after `FIND_PACKAGE(Boost...`.

Reply:
Have you tried compiling it recently, with current codebase ?

I successfully compiled the the main project's `develop` branch with `fc` being this branch, using Gitian through Docker, without the need to move `CheckLibcxxAtomic`.

Reply:
Hm, yes, seems to be working now.
I wonder why - `CheckLibcxxAtomic` compiles a test program that uses `boost::lockfree` - how can that work before boost has been detected?

Reply:
Looks like you were right.

I have moved `CheckLibcxxAtomic` after `find_package(Boost...)`, because in some cases it did not work, specifically when working with MinGW compiler.

Still, I can't explain the behaviour. I thought that CMake might do some multi-pass magic, but it has probably something to do with default compiler paths and also with the fact that the main project runs `find_package(Boost...)` before `CheckLibcxxAtomic`.